articles/tutorials/building_2d_games/16_service_container/index.md

```diff
@@ -0,0 +1,482 @@
+
+While these are different concepts, they often work together in practice. Many game systems are implemented as game components that also register themselves as services.
+
+### When to Use Game Components vs Services
+
+Use **Game Components** when you need:
+
+- Functionality that requires regular updates each frame.
+- Self-contained systems with their own update and/or draw logic.
+- Organization of different game subsystems.
+
+Use **Services** when you need:
+
+- Global access to a resource or system.
+- To decouple components that need to interact.
+- To share references without passing them explicitly through constructors.
+
+### When to Register a Game Component as a Service
+
+Not all game components need to be services, and not all services need to be game components. Here are some guidelines:
+
+**Register a game component as a service when:**
+
+- Multiple other components or systems need to access its functionality.
+- You want the flexibility to swap implementations later.
+- The component provides functionality that's useful throughout your game.
+
+**Keep a game component without registering it as a service when:**
+
+- It's self-contained and doesn't need to be accessed by other components.
+- It only affects a specific part of your game.
+- It doesn't provide functionality that other components need to use.
+
+## The Game Services Property
+
+At the core of MonoGame's service system is the [**Game.Services**](xref:Microsoft.Xna.Framework.Game.Services) property, which provides access to the game's [**GameServiceContainer**](xref:Microsoft.Xna.Framework.GameServiceContainer). This container has two primary methods:
+
+1. [**AddService**](xref:Microsoft.Xna.Framework.GameServiceContainer.AddService(System.Type,System.Object)): Registers a service with the container
+2. [**GetService**](xref:Microsoft.Xna.Framework.GameServiceContainer.GetService(System.Type)): Retrieves a service from the container.
+
+### Adding a Service
+
+When adding a service using the [**AddService**](xref:Microsoft.Xna.Framework.GameServiceContainer.AddService(System.Type,System.Object)) method, you need to specify:
+
+1. The service type (usually an interface).
+2. The service implementation (usually a concrete class).
+
+While you can register services by their concrete type, it's considered best practice to register them by their interface instead. This allows you to swap implementations without changing the consuming code.
+
+For example, you might register an `ExampleService` like this:
+
+```cs
+// Define an interface
+public interface IExampleService
+{
+    void DoSomething();
+    // Interface members ...
+}
+
+// Create a concrete implementation of the interface.
+public class ExampleService : GameComponent, IExampleService
+{
+    // Implementation. ...
+}
+
+// Register it as a service
+Game.Services.AddService<IExampleService>(new ExampleService(Game));
+```
+
+### Retrieving a Service
```

Comment:
Sounds daft, but the first question that pops into my mind, "why would I want to retrieve it?"

Kinda needs a flow around how a service would realistically be used in a game.